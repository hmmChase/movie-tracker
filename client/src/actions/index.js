export const addMovies = movies => ({
  type: 'ADD_MOVIES',
  movies
});

export const toggleLogin = () => ({
  type: 'TOGGLE_LOGIN'
});

export const storeUserData = ({ id, name }) => ({
  type: 'STORE_USER_DATA',
  id,
  name
});

export const loadFavorites = favorites => ({
  type: 'LOAD_FAVORITES',
  favorites
});