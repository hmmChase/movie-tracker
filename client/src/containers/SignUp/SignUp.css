.SignUp p {
  margin-bottom: 2rem;
  color: lightblue;
  font-family: 'Permanent Marker';
  font-size: 2rem;
  letter-spacing: 1.5px;
  text-shadow: 2px 2px #2362ce; }

.SignUp button {
  display: flex;
  margin: 2rem auto; }

.inputField {
  width: 250px;
  height: 40px;
  padding-left: 20px;
  border-radius: 20px;
  margin-right: 30px;
  font-size: 1.2rem;
  letter-spacing: 2px;
  background-color: rgba(118, 179, 181, 0.845);
  word-spacing: 3px; }
