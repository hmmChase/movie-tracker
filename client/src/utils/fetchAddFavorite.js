export const fetchAddFavorite = async (movie, user_id) => {
  try {
    const url = '/api/users/favorites/new';
    const options = {
      method: 'POST',
      body: JSON.stringify({
        ...movie,
        user_id
      }),
      headers: { 'Content-Type': 'application/json' }
    };
    const response = await fetch(url, options);

    if (!response.ok) {
      throw new Error(`${response.status}`);
    }
    return response;
  } catch (error) {
    throw new Error(`Network request failed. (error: ${error.message})`);
  }
};
