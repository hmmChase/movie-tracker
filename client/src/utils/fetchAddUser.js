export const fetchAddUser = async user => {
  try {
    const url = '/api/users/new';
    const options = {
      method: 'POST',
      body: JSON.stringify(user),
      headers: { 'Content-Type': 'application/json' }
    };
    const response = await fetch(url, options);
    if (!response.ok) {
      throw new Error(`${response.status}`);
    }
    return response;
  } catch (error) {
    throw new Error(`Network request failed. (error: ${error.message})`);
  }
};
