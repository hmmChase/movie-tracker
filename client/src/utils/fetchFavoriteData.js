import { doFetch } from './doFetch';

export const fetchFavoriteData = async user_id => {
  const url = `/api/users/${user_id}/favorites`;
  const favoritesList = await doFetch(url);
  return favoritesList.data;
};
