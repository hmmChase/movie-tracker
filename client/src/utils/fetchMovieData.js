import { apiKey } from '../private/apiKey';
import { doFetch } from './doFetch';

export const fetchMovieData = async () => {
  const url = 'https://api.themoviedb.org/3/movie/now_playing?api_key=';
  const dirtyMovieData = await doFetch(url + apiKey);
  return cleanMovieData(dirtyMovieData.results);
};

export const cleanMovieData = movieData => {
  return movieData.map(movie => ({
    movie_id: movie.id,
    title: movie.title,
    poster_path: movie.poster_path,
    release_date: movie.release_date,
    vote_average: movie.vote_average,
    overview: movie.overview
  }));
};
