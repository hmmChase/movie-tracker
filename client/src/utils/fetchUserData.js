import { doFetch } from './doFetch';

export const fetchUserData = async () => await doFetch('/api/users');
