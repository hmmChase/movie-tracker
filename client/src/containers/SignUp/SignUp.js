import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { fetchUserData } from '../../utils/fetchUserData';
import { fetchAddUser } from '../../utils/fetchAddUser';
import { toggleLogin, storeUserData } from '../../actions';
import './SignUp.css';

export class SignUp extends Component {
  constructor() {
    super();

    this.state = {
      name: '',
      email: '',
      password: ''
    };
  }

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();
    const foundUser = await this.findUser();
    foundUser ? alert('This email is already in use') : this.addUser();
  };

  addUser = async () => {
    await fetchAddUser(this.state);
    await fetchUserData();
    const foundUser = await this.findUser();
    this.props.storeUserData(foundUser);
    this.props.toggleLogin();
    this.props.history.push('/');
  };

  findUser = async () => {
    const userData = await fetchUserData();
    return userData.data.find(
      user => user.email === this.state.email.toLowerCase()
    );
  };

  render() {
    return (
      <div className="SignUp">
        <p>Sign up for a new account</p>
        <form onSubmit={this.handleSubmit}>
          <input
            type="text"
            name="name"
            placeholder="Enter a Name"
            value={this.state.name}
            onChange={this.handleChange}
            className="inputField"
          />
          <input
            type="text"
            name="email"
            placeholder="Enter an Email"
            value={this.state.email}
            onChange={this.handleChange}
            className="inputField"
          />
          <input
            type="password"
            name="password"
            placeholder="Enter a Password"
            value={this.state.password}
            onChange={this.handleChange}
            className="inputField"
          />
          <button className="signUpButton" />
        </form>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  toggleLogin: () => dispatch(toggleLogin()),
  storeUserData: userData => dispatch(storeUserData(userData))
});

SignUp.propTypes = {
  toggleLogin: PropTypes.func.isRequired,
  storeUserData: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
};

export default withRouter(connect(null, mapDispatchToProps)(SignUp));
