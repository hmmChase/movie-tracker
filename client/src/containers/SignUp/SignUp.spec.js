import React from 'react';
import { shallow } from 'enzyme';
import { SignUp, mapDispatchToProps } from './SignUp';
import { toggleLogin, storeUserData } from '../../actions/';
import { fetchAddUser } from '../../utils/fetchAddUser';
jest.mock('../../utils/fetchAddUser');
import { fetchUserData } from '../../utils/fetchUserData';
jest.mock('../../utils/fetchUserData');

describe('SignUp', () => {
  let signUp;
  let mockProps;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {
      toggleLogin: jest.fn(),
      storeUserData: jest.fn(),
      history: { push: jest.fn() }
    };
    signUp = shallow(<SignUp {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(signUp).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('sets state.name with value of name input', () => {
      const mockEvent = {
        target: {
          name: 'name',
          value: 'mock name'
        }
      };
      signUp.instance().handleChange(mockEvent);
      const results = { name: 'mock name', email: '', password: '' };

      expect(signUp.state()).toEqual(results);
    });

    it('sets state.email on change of the email input', () => {
      const mockEvent = {
        target: {
          name: 'email',
          value: 'mock@email.com'
        }
      };
      const results = { name: '', email: 'mock@email.com', password: '' };
      signUp.instance().handleChange(mockEvent);
      expect(signUp.state()).toEqual(results);
    });

    it('sets state.password on change of the password input', () => {
      const mockEvent = {
        target: {
          name: 'password',
          value: 'password'
        }
      };
      signUp.instance().handleChange(mockEvent);
      const results = { name: '', email: '', password: 'password' };
      expect(signUp.state()).toEqual(results);
    });
  });

  describe('handleSubmit', () => {
    it('calls findUser', () => {
      const mockEvent = { preventDefault: jest.fn() };
      const findUser = (signUp.instance().findUser = jest.fn());
      signUp.instance().addUser = jest.fn();
      signUp.instance().handleSubmit(mockEvent);

      expect(findUser).toHaveBeenCalledTimes(1);
    });

    it('calls alert if findUser is true', async () => {
      const mockEvent = { preventDefault: jest.fn() };
      window.alert = jest.fn();
      signUp.instance().findUser = jest.fn().mockImplementation(() => true);
      await signUp.instance().handleSubmit(mockEvent);

      expect(window.alert).toHaveBeenCalledTimes(1);
      expect(window.alert).toHaveBeenCalledWith('This email is already in use');
    });

    it('calls addUser if findUser is false', async () => {
      const mockEvent = { preventDefault: jest.fn() };
      signUp.instance().findUser = jest.fn().mockImplementation(() => false);
      const addUser = (signUp.instance().addUser = jest.fn());
      await signUp.instance().handleSubmit(mockEvent);

      expect(addUser).toHaveBeenCalledTimes(1);
    });
  });

  describe('addUser', () => {
    it('calls fetchAddUser with correct params', () => {
      const mockState = { email: '', name: '', password: '' };
      signUp.instance().findUser = jest.fn();
      signUp.instance().addUser();

      expect(fetchAddUser).toHaveBeenCalledTimes(1);
      expect(fetchAddUser).toHaveBeenCalledWith(mockState);
    });

    it('calls fetchUserData', async () => {
      signUp.instance().findUser = jest.fn();
      await signUp.instance().addUser();

      expect(fetchUserData).toHaveBeenCalledTimes(1);
    });
  });

  describe('findUser', () => {
    it('calls fetchUserData', () => {
      signUp.setState({
        email: 'dave@aol.com'
      });
      const userData = {
        data: [{ id: 1, name: 'dave', email: 'dave@aol.com' }]
      };
      fetchUserData.mockImplementation(() => userData);
      signUp.instance().findUser();

      expect(fetchUserData).toHaveBeenCalledTimes(1);
    });

    it('returns a user object', async () => {
      signUp.setState({
        email: 'dave@aol.com'
      });
      const userData = {
        data: [{ id: 1, name: 'dave', email: 'dave@aol.com' }]
      };
      const expected = { email: 'dave@aol.com', id: 1, name: 'dave' };
      fetchUserData.mockImplementation(() => userData);

      expect(await signUp.instance().findUser()).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch on toggleLogin', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = toggleLogin();
      mappedProps.toggleLogin();

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('calls dispatch with the correct params on storeUserData', () => {
      const mockDispatch = jest.fn();
      const userData = {};
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = storeUserData(userData);
      mappedProps.storeUserData(userData);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
});
