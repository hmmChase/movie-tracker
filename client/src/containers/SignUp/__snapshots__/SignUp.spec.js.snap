// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    history={
        Object {
            "push": [Function],
          }
    }
    storeUserData={[Function]}
    toggleLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Sign up for a new account
</p>,
        <form
          onSubmit={[Function]}
>
          <input
                    className="inputField"
                    name="name"
                    onChange={[Function]}
                    placeholder="Enter a Name"
                    type="text"
                    value=""
          />
          <input
                    className="inputField"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter an Email"
                    type="text"
                    value=""
          />
          <input
                    className="inputField"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter a Password"
                    type="password"
                    value=""
          />
          <button
                    className="signUpButton"
          />
</form>,
      ],
      "className": "SignUp",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign up for a new account",
        },
        "ref": null,
        "rendered": "Sign up for a new account",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="inputField"
              name="name"
              onChange={[Function]}
              placeholder="Enter a Name"
              type="text"
              value=""
/>,
            <input
              className="inputField"
              name="email"
              onChange={[Function]}
              placeholder="Enter an Email"
              type="text"
              value=""
/>,
            <input
              className="inputField"
              name="password"
              onChange={[Function]}
              placeholder="Enter a Password"
              type="password"
              value=""
/>,
            <button
              className="signUpButton"
/>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "inputField",
              "name": "name",
              "onChange": [Function],
              "placeholder": "Enter a Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "inputField",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Enter an Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "inputField",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Enter a Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signUpButton",
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Sign up for a new account
</p>,
          <form
            onSubmit={[Function]}
>
            <input
                        className="inputField"
                        name="name"
                        onChange={[Function]}
                        placeholder="Enter a Name"
                        type="text"
                        value=""
            />
            <input
                        className="inputField"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter an Email"
                        type="text"
                        value=""
            />
            <input
                        className="inputField"
                        name="password"
                        onChange={[Function]}
                        placeholder="Enter a Password"
                        type="password"
                        value=""
            />
            <button
                        className="signUpButton"
            />
</form>,
        ],
        "className": "SignUp",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign up for a new account",
          },
          "ref": null,
          "rendered": "Sign up for a new account",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="inputField"
                name="name"
                onChange={[Function]}
                placeholder="Enter a Name"
                type="text"
                value=""
/>,
              <input
                className="inputField"
                name="email"
                onChange={[Function]}
                placeholder="Enter an Email"
                type="text"
                value=""
/>,
              <input
                className="inputField"
                name="password"
                onChange={[Function]}
                placeholder="Enter a Password"
                type="password"
                value=""
/>,
              <button
                className="signUpButton"
/>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "inputField",
                "name": "name",
                "onChange": [Function],
                "placeholder": "Enter a Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "inputField",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Enter an Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "inputField",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Enter a Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "signUpButton",
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
